src/bundles/csg/functions.ts

```diff
@@ -315,324 +486,75 @@ export function intersect(a: Shape, b: Shape): Shape {
 }
 
 /**
- * Scales the shape in the x, y and z direction with the specified factor,
- * ranging from 0 to infinity.
- * For example scaling the shape by 1 in x, y and z direction results in
+ * Scales the shape in the x, y and z direction with the specified factor.
+ * Factors must be non-zero.

```

Comment:
Why must factors be non-zero? Strictly speaking, shouldn't it be technically possible to scale an entity by a factor of 0 (making it infinitesimally small)?

If it is to get around an error, I don't think this is the best way to do it.

Reply:
The JSCAD code (node_modules\@jscad\modeling\src\operations\transforms\scale.js) currently does not even allow negative values and will throw an error if factors are <=0. We decided to add a <=0 check in function.ts to avoid revealing the JSCAD implementation

Reply:
This sounds like we are artificially enforcing that factors are positive simply to get around this limitation/error of the JSCAD library. It's not about revealing implementation; what I meant was more of a question of "should we support scaling by non-positive factors?".

If we can define scaling by a negative number to be _"flip in the corresponding axes that are scaled by a negative factor, and then scale by the absolute value of the factors" then why can't we just implement it that way in the code? Artificially capping it to >= 0 just because it will through an error otherwise does not seem like the best thing to do. Is there any reason (perhaps pedagogical/etc.) that it was not done this way?

Likewise, for scaling with a 0 factor.

Reply:
While it is possible to support negative scaling by patching the JSCAD code, Prof has said that he does not support the inclusion of such a functionality.

"I don't support the inclusion of negative scaling. It is not often used in 3D modelling. Would rather have a separate flip function to do that. Since all the primitive shapes we provide have some symmetry such that flipping can be achieved with rotation, a separate flip function or negative scaling is not necessary."

Reply:
Okay sure, if that's the reason then I guess that's fine. Thanks for the clarification!

Reply:
Hi @RichDom2185, when trying to merge this with the master branch, I get a syntax error when importing the CSG module in the frontend. I suspect this has something to do with the @jscad/modeling version, however, I have not found a way to resolve the merge conflict without causing this error. Could you please look into this? Thanks!

Reply:
What exactly is the command you are using to merge `master` into this branch? Your force pushes seem to be causing weird things in the GitHub PR list of changes.

Reply:
I will try to troubleshoot. Essentially, the module is done right? All that's left is to merge `master`?

Reply:
Yep, all that's left is to merge `master`

Reply:
Can you create a new branch with the contents of this PR just before your most recent force-pushed? (This one below)

<img width="923" alt="image" src="https://github.com/source-academy/modules/assets/34370238/df02b8f2-15fd-4fdb-909c-b697705a79d7">


What was the rationale of the force-push btw?

Reply:
Nevermind, was able to dig the commit out of GitHub's servers. See #213.